src/main.rs

```diff
@@ -129,174 +129,195 @@ pub async fn do_main() -> Result<()> {
 		)
 		.unwrap();
 	});
-
 	if let Ok((peer_id, addrs)) = self_info_rx.recv() {
 		log::info!("IPFS backed application client: {}\t{:?}", peer_id, addrs);
-	}
-
-	let block_header = rpc::get_chain_header(&cfg.full_node_rpc).await?;
+	};
+	let rpc_url = match rpc::check_http(cfg.full_node_rpc).await {
+		Ok(a) => a,
+		Err(e) => return Err(e),
+	};
+	let rpc_: &str = &*rpc_url;
```

Comment:
Is it possible to reduce this to something like:

`let rpc_url = match rpc::check_http(cfg.full_node_rpc).await?.clone();`?